climbing-stairs/bhyun-kim.py

```diff
@@ -0,0 +1,39 @@
+"""
+70. Climbing Stairs
+https://leetcode.com/problems/climbing-stairs/description/
+
+Solution:
+    This is a combinatorial problem that allows duplicates.
+    We can use the formula for combinations to solve this problem.
+    Let's suppose that n is the number of steps, and k is the number of 2-steps.
+    Then, the number of ways to climb the stairs is n!/((n-2k)!k!).
+    We can iterate through all possible values of k, and calculate the number of ways to climb the stairs.
+
+    1. Create a dictionary to store the factorials of numbers from 0 to n.
+    2. Calculate the factorials of numbers from 0 to n.
+    3. Iterate through all possible values of k from 0 to n//2.
+
+
+Time complexity: O(n)
+Space complexity: O(n)   
+"""
+
+from collections import defaultdict
+
+
+class Solution:
+    def climbStairs(self, n: int) -> int:
+        factorials = defaultdict(int)
+        factorials[0] = 1
+
+        fact = 1
+        for i in range(1, n + 1):
+            fact = fact * i
+            factorials[i] = fact
+
+        output = 0
+
+        for i in range(n // 2 + 1):
+            num_ways = factorials[n - i] / factorials[n - (i * 2)] / factorials[i]
```

Comment:
```suggestion
            num_ways = factorials[n - i] // factorials[n - (i * 2)] // factorials[i]
```
요 부분이 조금 걸리네용. 수학적으로 조합이나 순열을 썼을 때 실수보단 정수가 나오는게 맞는거 같은데 어떻게 생각하시는지 궁금합니다!

Reply:
또한 이 코드도 물론 잘 돌아가겠지만, 공간복잡도를 O(1)로 줄일 수 있는 방법도 있으니 한 번 고민해보시는걸 추천드려용 😃

Reply:
> 요 부분이 조금 걸리네용. 수학적으로 조합이나 순열을 썼을 때 실수보단 정수가 나오는게 맞는거 같은데 어떻게 생각하시는지 궁금합니다!

안그래도 정수로 리턴해야하는데 실수가 나와서 정답 제출할 때 걸리더라구요! 이 방법이 더 쉬웠겠네요. 감사드립니다.